sdv/multi_table/base.py

```diff
@@ -217,7 +217,8 @@ def auto_assign_transformers(self, data):
         """
         for table_name, table_data in data.items():
             self._validate_table_name(table_name)
-            self._table_synthesizers[table_name].auto_assign_transformers(table_data)
+            synthesizer = self._table_synthesizers[table_name]
+            self._skip_foreign_key_transformations(synthesizer, table_name, table_data)
```

Comment:
This method calls `self.auto_assign_transformers` so I think this will just recurse forever. I think instead we could move the logic inside of `_skip_foreign_key_transformations` that sets the foreign key's transformers to None inside of this method, and just call this method everywhere that `_skip_foreign_key_transformations` is currently called